src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMetaboIdent.cpp

```diff
@@ -485,260 +475,6 @@ namespace OpenMS
-      vector<FeatureGroup> no_overlaps;
-      for (const FeatureGroup& group : overlap_groups)
-      {
-        if (hasOverlappingFeature_(feat, group, feature_bounds))
-        {
-          current_overlaps.push_back(group);
-        }
-        else
-        {
-          no_overlaps.push_back(group);
-        }
-      }
-      if (current_overlaps.empty()) // make new group for current feature
-      {
-        FeatureGroup new_group(1, &(feat));
-        no_overlaps.push_back(new_group);
-      }
-      else // merge all groups that overlap the current feature, then add it
-      {
-        FeatureGroup& merged = current_overlaps.front();
-        for (vector<FeatureGroup>::const_iterator group_it =
-               ++current_overlaps.begin(); group_it != current_overlaps.end();
-             ++group_it)
-        {
-          merged.insert(merged.end(), group_it->begin(), group_it->end());
-        }
-        merged.push_back(&feat);
-        no_overlaps.push_back(merged);
-      }
-      overlap_groups.swap(no_overlaps);
-    }
-  }
-
-  /// Resolve overlapping features by picking the best and removing all others
-  void FeatureFinderAlgorithmMetaboIdent::resolveOverlappingFeatures_(FeatureGroup& group,
-                                   const FeatureBoundsMap& feature_bounds)
-  {
-    if (debug_level_ > 0)
-    {
-      String msg = "Overlapping features: ";
-      for (FeatureGroup::const_iterator it = group.begin(); it != group.end();
-           ++it)
-      {
-        if (it != group.begin())
-        {
-          msg += ", ";
-        }
-        msg += String((*it)->getMetaValue("PeptideRef")) + " (RT " +
-          String(float((*it)->getRT())) + ")";
-      }
-      OPENMS_LOG_DEBUG << msg << endl;
-    }
-
-    Feature* best_feature = nullptr;
-    while (!group.empty())
-    {
-      double best_rt_delta = numeric_limits<double>::infinity();
-      // best feature is the one with min. RT deviation to target:
-      for (FeatureGroup::const_iterator it = group.begin(); it != group.end();
-           ++it)
-      {
-        double rt_delta = abs(double((*it)->getMetaValue("rt_deviation")));
-        if ((rt_delta < best_rt_delta) ||
-            ((rt_delta == best_rt_delta) && ((*it)->getIntensity() >
-                                             best_feature->getIntensity())))
-        {
-          best_rt_delta = rt_delta;
-          best_feature = *it;
-        }
-        else if ((rt_delta == best_rt_delta) && ((*it)->getIntensity() ==
-                                                 best_feature->getIntensity()))
-        {
-          // are the features the same? (@TODO: use "Math::approximatelyEqual"?)
-          if (((*it)->getRT() == best_feature->getRT()) &&
-              ((*it)->getMZ() == best_feature->getMZ()))
-          {
-            // update annotations:
-            // @TODO: also adjust "formula" and "expected_rt"?
-            String label = best_feature->getMetaValue("label");
```

Comment:
@hendrikweisser is this part relevant? In my test data, this case never occurred afaict but I guess this happens from time to time?
@jpfeuffer for the sake of completeness should add another "merge" functor that in cases overlapping features are equal allow to merge them?

Reply:
Yes maybe it is good to offer this. Although an MS2 could be in a non overlapping region.